lib/dogma/rules/predicate_name.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Dogma.Rules.PredicateName do
+  @moduledoc """
+  A rule that disallows tautological predicate names.
+  """
+
+  @behaviour Dogma.Rule
+  alias Dogma.Script
+  alias Dogma.Error
+
+  def test(script) do
+    script |> Script.walk( &check_node(&1, &2) )
+  end
+
+  defp check_node({:def, _, [{name, meta, _}|_]} = node, errors) do
+    test_predicate(name, meta, node, errors)
+  end
+  defp check_node({:defp, _, [{name, meta, _}|_]} = node, errors) do
+    test_predicate(name, meta, node, errors)
+  end
+  defp check_node(node, errors) do
+    {node, errors}
+  end
+
+  defp test_predicate(line) do
+    Regex.run(~r{(is|has)_(\w+)\?}, line)
+  end
+
+  defp tautological_predicate?(line) do
+    !!(test_predicate(line))
```

Comment:
Having a function that does nothing but cast to boolean doesn't really add anything in a language with duck typing, so let's remove this, and just use `test_predicate/1` in its place.